src/goto-cc/CMakeLists.txt

```diff
@@ -29,4 +29,14 @@ target_link_libraries(goto-cc goto-cc-lib)
 
 if(WIN32)
   set_target_properties(goto-cc PROPERTIES OUTPUT_NAME goto-cl)
+else()
+  # regression/goto-gcc requires an _executable_ target; this cruel hack will
+  # first build the executable and then replace it by a symbolic link so as to
+  # save some disk space
+  add_executable(goto-gcc goto_cc_main.cpp)
```

Comment:
I suppose it would be good for @reuk to have a look at this work around. 

To check I understand, it is compiling `goto_cc_main.cpp` -> `goto-gcc` then deleting it by replacing it with a symlink to `goto-cc`? 

How come this is only necessary on Linux. Or rather: why can't we do `set_target_properties(goto-cc PROPERTIES OUTPUT_NAME goto-gcc)`. I realise you've spent quite a few cycles iterating to get something working, so it would be good to expand the comment to explain why it is necessary.

Reply:
Yes, I'd also like to know *why* regression/goto-gcc requires an executable target. Does `TARGET_FILE:goto-gcc` not work otherwise?

Reply:
Another option is to set an `alternative path/name` property on `goto-cc` (set to the path of goto-gcc` and then to query it using `$<TARGET_PROPERTY:tgt,prop>` in regression/goto-gcc

Reply:
@reuk here's what I get:
```
CMake Error at regression/CMakeLists.txt:7 (add_test):
  Error evaluating generator expression:

    $<TARGET_FILE:goto-gcc>

  Target "goto-gcc" is not an executable or library.
```
Looking at CMake's documentation, this is maybe not surprising:
"$<TARGET_FILE:tgt>
    Full path to main file (.exe, .so.1.2, .a) where tgt is the name of a target."

I'm now trying some string rewriting, which I'd say is a variant on the properties approach? Comments solicited!